Code/GraphMol/ChemTransforms/MolFragmenter.cpp

```diff
@@ -591,6 +594,227 @@ ROMol *fragmentOnBRICSBonds(const ROMol &mol) {
+                nswaps++;
+            }
+        }
+    }
+    return nswaps;
+}
+
+// Simple bookkeeping class to bond attachments and handle stereo
+struct ZipBond {
+    Atom *a;       // atom being bonded
+    Atom *a_dummy; // Labelled atom, i.e. [*:1]-C  will bond the C to something
+    Atom *b;       // atom being bonded
+    Atom *b_dummy; // Labelled atom, i.e. [*:1]-O will bond the O to something
+    
+    // Mark the original order of the nbr atoms including the dummy
+    //  The goal is to copy the dummy chiral order over to the
+    //  atom being bonded
+    void mark(Atom *chiral_atom, Atom *dummy_atom, Atom *new_atom) const {
+        if(!chiral_atom->getChiralTag()) {
+            return;
+        }
+        std::string mark = "mark_" + std::to_string(chiral_atom->getIdx());
+        int order = 0;
+        auto &m = chiral_atom->getOwningMol();
+        for(auto nbrIdx : boost::make_iterator_range(m.getAtomNeighbors(chiral_atom))) {
+            m[nbrIdx]->setProp<int>(mark, order);
+            ++order;
+        }
+        new_atom->setProp<int>(mark, dummy_atom->getProp<int>(mark));
+    }
+    
+    // Backup the original chirality
+    void mark_chirality() const {
+        mark(a, a_dummy, b);
+        mark(b, b_dummy, a);
+    }
+    
+    // bond a<->b for now only use single bonds
+    //  XXX FIX ME take the highest bond order.
+    bool bond(RWMol &newmol) const {
+        if (!a || !b) {
+            BOOST_LOG(rdWarningLog) << "Incomplete atom labelling, cannot make bond" << std::endl;
+            return false;
+        }
+        if (!a->getOwningMol().getBondBetweenAtoms(a->getIdx(), b->getIdx())) {
+            assert (&a->getOwningMol() == &newmol);
+            newmol.addBond(a, b, Bond::BondType::SINGLE);
+        }
+        return true;
+    }
+    
+    // Restore the atom's chirality by comparing the original order
+    //  to the current
+    void restore(Atom *chiral_atom) const {
+        if(!chiral_atom->getChiralTag()) {
+            return;
+        }
+        std::string mark = "mark_" + std::to_string(chiral_atom->getIdx());
+        //std::vector<unsigned int> orders1;
+        std::vector<unsigned int> orders2;
+        auto &m = chiral_atom->getOwningMol();
+        for(auto nbrIdx : boost::make_iterator_range(m.getAtomNeighbors(chiral_atom))) {
+            orders2.push_back(m[nbrIdx]->getProp<int>(mark));
+        }
+        if(num_swaps_to_interconvert(orders2) % 2 == 1) {
+            chiral_atom->invertChirality();
+        }
+    }
+    
+    // Restore the marked chirality
+    void restore_chirality(std::set<Atom *> &already_checked) const {
+        if(already_checked.find(a) == already_checked.end()) {
+            restore(a);
+            already_checked.insert(a);
+        }
+        if(already_checked.find(b) == already_checked.end()) {
+            restore(b);
+            already_checked.insert(b);
+        }
+    }
+};
+}
+
+std::unique_ptr<ROMol> molzip(
```

Comment:
seems like this would be significantly easier, on at least one side, if you take the bond to the dummy atom in molA and use `setEndAtomIdx()` (or `setBeginAtomIdx()` if appropriate) to point it to the appropriate atom that you've added from `molB`?

Reply:
Oh, that's cute.  I would never have thought of. that.  Actually, if you do it this way, you don't have to actually check chirality, do you?

Reply:
I believe that is correct

Reply:
It doesn't appear to work as that call doesn't appear to update the molgraph.

Reply:
ah! minor details!
Sorry, I should have thought about that when I made the suggestion